### PR TITLE
fix: show "Ferry from" if it's ferry and not boat

### PR DESCRIPTION
**Pull request by adriansberg**
This PR fixes a mislabeling issue, ensuring the correct information is shown for transportation mode.

#### Changes Include:
-  Updated logic to correctly display "Ferry from" when the transportation mode is a ferry, not a boat.

**Impact**: This change improves the display information integrity, enhancing user understanding and experience.

Closes https://github.com/AtB-AS/kundevendt/issues/15996

### CODE REVIEW

File: src/translations/pages/assistant.ts [hunk 1]
```diff
@@ -68,10 +76,17 @@ export const Assistant = {
               `Tog frå ${place}`,
             );
           case 'water':
+            if (isSubmodeBoat(submode)) {
```
[the remark this answers is not in the document]
Comment (adriansberg) [reply src/translations/pages/assistant.ts:79]: I'll refactor this. IIRC the app uses the dictionary.ts for mode translations. Can check how they do this line.

[the remark this answers is not in the document]
Comment (mikaelbr) [reply src/translations/pages/assistant.ts:79]: This is one of the cases where I think maybe the App can be too pragmatic. 🤔 Generally, I would say that translation files should have minimal, if any, logic

[the remark this answers is not in the document]
Comment (adriansberg) [reply src/translations/pages/assistant.ts:79]: Do you think it's okay to pass place to the translation, or should that be handled outside as well?

2023-12-13 at 10 19 14

[the remark this answers is not in the document]
Comment (mikaelbr) [reply src/translations/pages/assistant.ts]: No, that is a pattern that is used around, and that isn't really logic. So thinking this is fine

**Review by mikaelbr — APPROVED** [review]
Looks better I think!

**Pull request merged**